tls/s2n_ktls.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include <stdbool.h>
+
+#include "utils/s2n_safety.h"
+
+struct s2n_connection;
+
+/* A set of kTLS configurations representing the combination of sending
+ * and receiving.
+ */
+typedef enum {
```

Comment:
Is this intended to be a public enum? If so, then I'm not a fan of also using its fields as bitflags / assuming they're bitflags.